### PR TITLE
Align severity handling for CLI exit codes

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/commands/AdvisorCommand.kt [hunk 1]
```diff
@@ -36,10 +36,12 @@ import com.github.ajalt.clikt.parameters.types.file
 
 import org.ossreviewtoolkit.advisor.Advisor
 import org.ossreviewtoolkit.cli.GlobalOptions
+import org.ossreviewtoolkit.cli.printSeverityStats
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/AdvisorCommand.kt:39]: Commit message typo: "iss**u**es"

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt [hunk 2]
```diff
@@ -280,16 +281,6 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate ORT re
...
-        val warningCount = counts[Severity.WARNING] ?: 0
-        val hintCount = counts[Severity.HINT] ?: 0
-
-        if (errorCount > 0 || warningCount > 0) {
-            println("Found $errorCount errors, $warningCount warnings, $hintCount hints.")
-            throw ProgramResult(2)
-        }
-
-        println("Found $hintCount hints only.")
+        if (printSeverityStats(counts, Severity.HINT)) throw ProgramResult(2)
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/commands/EvaluatorCommand.kt:284]: This reads like `ProgramResult(2)` is thrown if the stats could be printed, I'd prefer to find a better name for the function but don't have a good suggestion. Another option would be to split `printSeverityStats` and the `hasSevereIssues` check into separate functions.

**Review by mnonnenmacher — DISMISSED** [review]

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt:284]: Or how about simply moving the `throw ProgramResult(2)` into the function (as that's what all callers currently do) and not return anything?

File: cli/src/main/kotlin/OrtMain.kt [hunk 3]
```diff
@@ -71,6 +72,35 @@ data class GlobalOptions(
...
+
+    fun getSeverityCount(severity: Severity) =
+        counts.getOrDefault(severity, 0).also { hasSevereIssues = it > 0 && severity > threshold }
+
+    val hintCount = getSeverityCount(Severity.HINT)
+    val warningCount = getSeverityCount(Severity.WARNING)
+    val errorCount = getSeverityCount(Severity.ERROR)
+
+    println("Found $errorCount errors, $warningCount warnings, $hintCount hints.")
+    if (hasSevereIssues) println("There are severe issues.")
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt:90]: I went for something even more verbose.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt]: I did that now in the new iteration.

**Review by oheger-bosch — APPROVED** [review]

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 2 above)
Comment (mnonnenmacher) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt]: I still find it very unexpected that a function with the name "print..." has such a side effect, but it's merged now, so be it.

Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt]: I'm more than happy to make a follow-up PR to rename, it's not carved in stone. But the alignment of the exit code was more important to me right now. Do you have a proposal for a name by now?

**Pull request merged**